setup-utils/src/io/read.rs

```diff
@@ -62,15 +62,7 @@ impl<R: Read> Deserializer for R {
     ) -> Result<G> {
         let point = match compression {
             UseCompression::Yes => G::deserialize(self)?,
-            UseCompression::No => {
-                if check_for_correctness == CheckForCorrectness::OnlyNonZero
-                    || check_for_correctness == CheckForCorrectness::No
-                {
-                    G::deserialize_unchecked(self)?
-                } else {
-                    G::deserialize_uncompressed(self)?
-                }
-            }
+            UseCompression::No => G::deserialize_uncompressed(self)?,
```

Comment:
unchecked variant is no longer required, or it's handled by the new uncompressed function?